src/magic_folder/web.py

```diff
@@ -313,17 +317,35 @@ def render_GET(self, request):
         """
         Render a list of Magic Folders and some of their details, encoded as JSON.
         """
-        _application_json(request)
-        def magic_folder_details():
+        include_secret_information = int(request.args.get("include_secret_information", [0])[0])
+        _application_json(request)  # set reply headers
+
+        def get_folder_info(name, mf):
+            info = {
+                u"name": name,
+                u"author": {
+                    u"name": mf.author.name,
+                    u"verify_key": mf.author.verify_key.encode(Base32Encoder),
+                },
+                u"stash_path": mf.stash_path.path,
+                u"magic_path": mf.magic_path.path,
+                u"poll_interval": mf.poll_interval,
+                u"is_admin": mf.is_admin(),
+            }
+            if include_secret_information:
+                info[u"author"][u"signing_key"] = mf.author.signing_key.encode(Base32Encoder)
+                info[u"collective_dircap"] = mf.collective_dircap.encode("ascii")
+                info[u"upload_dircap"] = mf.upload_dircap.encode("ascii")
```

Comment:
I think JSON is happier with unicode than bytes ... although since caps are always ascii I don't know how much it matters here.  Still, there's no "bytes" type in JSON.